pkg/gpu/ebpf/c/runtime/gpu.c

```diff
@@ -208,4 +211,115 @@ int BPF_URETPROBE(uretprobe__cudaSetDevice) {
     return 0;
 }
 
+SEC("uprobe/cudaEventRecord")
+int BPF_UPROBE(uprobe__cudaEventRecord, __u64 event, __u64 stream) {
+    __u64 pid_tgid = bpf_get_current_pid_tgid();
+    cuda_event_key_t key = { 0 };
+    cuda_event_value_t value = { 0 };
+
+    key.event = event;
+    key.pid = pid_tgid >> 32;
+
+    value.stream = stream;
+    value.last_access_ktime_ns = bpf_ktime_get_ns();
+
+    log_debug("cudaEventRecord: pid_tgid=%llu, event=%llu, stream=%llu", pid_tgid, event, stream);
+    bpf_map_update_with_telemetry(cuda_event_to_stream, &key, &value, BPF_ANY);
```

Comment:
add a comment indicating we add the event regardless of the return value of cudaEventRecord and the mapcleaner will clean it if the function actually fails